src/Symfony/Component/Yaml/Parser.php

```diff
@@ -227,7 +227,7 @@ private function doParse(string $value, int $flags)
                 }
 
                 if (!\is_string($key) && !\is_int($key)) {
-                    throw new ParseException(sprintf('%s keys are not supported. Quote your evaluable mapping keys instead.', is_numeric($key) ? 'Numeric' : 'Non-string'), $this->getRealCurrentLineNb() + 1, $this->currentLine);
+                    throw new ParseException(is_numeric($key) ? 'Numeric' : 'Non-string'.' keys are not supported. Quote your evaluable mapping keys instead.', $this->getRealCurrentLineNb() + 1, $this->currentLine);
```

Comment:
I suggest adding braces around the ternary operator, to make it more readable that the concatenation is not part of the else branch

Reply:
ok, it is not only about readability actually. That's the reason why tests are failing, because the concatenation is part of the else branch.